torch/distributed/fsdp/_common_utils.py

```diff
@@ -390,12 +390,16 @@ def f(module: torch.nn.Module, prefix: str, tree_level: int, *args, **kwargs):
                         submodule_name == "_fsdp_wrapped_module"
                         or submodule_name == "_dmp_wrapped_module"
                     ):
-                        warnings.warn(
-                            "An unexpected prefix is detected. This case "
-                            " should only happen when using DMP with FSDP. "
-                            f"prefix = {prefix}, "
-                            f"submodule_name = {submodule_name}"
-                        )
+                        if (
+                            not torch.distributed._functional_collectives.is_torchdynamo_compiling()
```

Comment:
nit: where should we move this util? seems like it will take on a bigger life than functional_collectives

Reply:
Not there, that's for sure! Theres 2 options I think:

1) A top level distributed only flag in utils or top level moduel `__init__` - like `torch.distributed.utils.is_compiling()`
2) Remove it entirely from distributed, move it to a torch level util, since its a generally useful sort of check import -> check flag pattern other modules will want as well.

Reply:
yea, i'd be in favor of 2.  i am wary of import cycles, that may be the very reason we (I?) put it there in the first place.  So probably don't try to do it in this PR unless you're feeling lucky :)